src/yunohost/certificate.py

```diff
@@ -639,6 +639,9 @@ def _prepare_certificate_signing_request(domain, key_file, output_folder):
     # Set the domain
     csr.get_subject().CN = domain
 
+    # Include xmpp-upload subdomain as subject alternate names
+    csr.add_extensions([crypto.X509Extension("subjectAltName", False, "DNS:xmpp-upload." + domain)])
+
```

Comment:
I am not sure you why have to take care of this one and the upper one.
Why do you need to touch these files?
LE certificates are working for all domains.
These files are here to handle the the CA (certificate authority) or this kind of stuff.

Reply:
This change does not affect the CA itself, it only affects every self-signed certificates generated with this CA.
Although let's encrypt certificates are much preferable, this change is still needed for people who don't use Let's encrypt yet.

Reply:
Would people have to install a LE certificat on `xmpp-upload` sub-domain to have the upload working. Is this change supposed to take care of this? Is `xmpp-upload` a default domain created for every domain on the instance? Or should it be the admin to create this domain name to have this feature? What’s the plan of this PR?

Reply:
It's all automatic (except for existing certificates which all need to be forcibly renewed). Every LE certificate will be valid for both "thedomain.tld" and "xmpp-upload.thedomain.tld". The same for self-signed certificates.
The only condition for LE is that the name xmpp-upload.thedomain.tld has a valid DNS record. But normally this is the case because yunohost's suggested configuration contains a wildcard DNS record. Nevertheless, I've added a CNAME record in the suggestion dns-config to make it more explicit.